lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -19,25 +22,34 @@ def status
     end
 
     def deploy_succeeded?
-      success = observed_generation == current_generation &&
-        desired_replicas == status_data['readyReplicas'].to_i &&
-        status_data['currentRevision'] == status_data['updateRevision']
+      success = observed_generation == current_generation
+
+      @pods.each do |pod|
+        success &= pod.definition["metadata"]["labels"]["controller-revision-hash"] == status_data['updateRevision']
+      end
+
       if update_strategy == ONDELETE
-        # Gem cannot monitor update since it doesn't occur until delete
         unless @success_assumption_warning_shown
           @logger.warn("WARNING: Your StatefulSet's updateStrategy is set to OnDelete, "\
-                       "which means updates will not be applied until its pods are deleted. "\
-                       "Consider switching to rollingUpdate.")
+                       "which means updates will not be applied until its pods are deleted.")
           @success_assumption_warning_shown = true
         end
+
+        if required_rollout == 'full'
+          success &= desired_replicas == status_data['readyReplicas'].to_i &&
+            desired_replicas == status_data['updatedReplicas'].to_i
+        end
       else
-        success &= desired_replicas == status_data['currentReplicas'].to_i
+        if required_rollout == 'full'
+          success &= status_data['currentRevision'] == status_data['updateRevision'] &&
+            desired_replicas == status_data['readyReplicas'].to_i &&
+            desired_replicas == status_data['currentReplicas'].to_i
```

Comment:
can we do something like this instead?
```
success &= (status_data['currentRevision'] == status_data['updateRevision']) && 
(desired_replicas == status_data['readyReplicas'].to_i == status_data['currentReplicas'].to_i)
```
or did you opt to write it this way for better readability?

Reply:
oh actually you could do 

```
desired_replicas == (status_data['readyReplicas'].to_i == status_data['currentReplicas'].to_i)
```

but it's not as clear to other people who might look at the code and the readability level goes down with that 🤔

Reply:
You can also do
```
success &= status_data['currentRevision'] == status_data['updateRevision']
success &= desired_replicas == status_data['readyReplicas'].to_i
success &= desired_replicas == status_data['currentReplicas'].to_i
```
No strong opinions